File: grooves-core/src/main/java/com/github/rahulsom/grooves/logging/Trace.java
package com.github.rahulsom.grooves.logging;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Marks a method as being used in tracing.
 * These methods calls are logged at trace level with information on nesting.
 *
 * @see IndentedLogging
 */
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface Trace {}

File: grooves-core/src/main/kotlin/com/github/rahulsom/grooves/logging/Trace.kt
package com.github.rahulsom.grooves.logging

/**
 * Marks a method as being used in tracing.
 * These methods calls are logged at trace level with information on nesting.
 *
 * @see [IndentedLogging]
 */
@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class Trace(
    val twoStep: Boolean
)